projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -2048,6 +2048,17 @@ public function rest_register_related_posts() {
 	 * @return array
 	 */
 	public function rest_get_related_posts( $object ) {
+		// If the Related Posts option is turned off, don't get the related posts.
+		$options = \Jetpack_Options::get_option( 'relatedposts', array() );
+		if ( empty( $options['enabled'] ) || ! $options['enabled'] ) {
+			return array();
+		}
+
+		// If the current post don't contain a Related Posts block, don't get the related posts.
+		if ( ! has_block( 'jetpack/related-posts' ) ) {
```

Comment:
Let's add a check for the function just in case.
```suggestion
		if ( function_exists( 'has_block' ) && ! has_block( 'jetpack/related-posts' ) ) {
```

Reply:
I assume it's the Core function so we don't need to check it 🤔

See https://developer.wordpress.org/reference/functions/has_block/.